packages/mobile/src/pincode/Pincode.tsx

```diff
@@ -2,35 +2,44 @@
  * This is a VIEW. We use it everwhere we need to show PIN pad
  * with an input, e.g. get/ensure/set pincode.
  */
-import Button, { BtnTypes } from '@celo/react-components/components/Button'
 import NumberKeypad from '@celo/react-components/components/NumberKeypad'
-import { fontStyles } from '@celo/react-components/styles/fonts'
-import { componentStyles } from '@celo/react-components/styles/styles'
-import React, { useCallback, useMemo } from 'react'
+import colors from '@celo/react-components/styles/colors.v2'
+import fontStyles from '@celo/react-components/styles/fonts.v2'
+import React, { useCallback } from 'react'
 import { ScrollView, StyleSheet, Text, View } from 'react-native'
-import PincodeTextbox from 'src/pincode/PincodeTextbox'
+import PincodeDisplay from 'src/pincode/PincodeDisplay'
+import { PIN_LENGTH } from 'src/pincode/utils'
 
 interface Props {
-  onPress: () => void
-  title: string
-  placeholder: string
-  buttonText: string
+  title?: string
+  errorText?: string
   maxLength?: number
-  isPinValid: (pin: string) => boolean
   pin: string
   onChangePin: (pin: string) => void
+  onCompletePin: (pin: string) => void
 }
 
-function Pincode(props: Props) {
-  const { maxLength, pin, onChangePin, onPress, title, placeholder, buttonText, isPinValid } = props
-
+function Pincode({
+  title,
+  errorText,
+  maxLength = PIN_LENGTH,
+  pin,
+  onChangePin,
+  onCompletePin,
+}: Props) {
   const onDigitPress = useCallback(
```

Comment:
I guess, we can remove `useCallback` here

Reply:
yep!